### PR TITLE
DCAC-254: Produce messages to item-group-ordered topic

### PR DESCRIPTION
**Pull request by VaughanJackson123**
* Tested and developed the production of `ItemGroupOrdered` messages to the `item-group-ordered` topic.

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/itemhandler/kafka/OrderMessageDefaultConsumerIntegrationTest.java [hunk 1]
```diff
@@ -59,6 +42,30 @@
...
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.Mockito.verify;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+import static uk.gov.companieshouse.itemhandler.kafka.ItemGroupOrderedFactory.FILING_HISTORY_DESCRIPTION;
+import static uk.gov.companieshouse.itemhandler.kafka.ItemGroupOrderedFactory.FILING_HISTORY_DESCRIPTION_VALUES;
+import static uk.gov.companieshouse.itemhandler.kafka.ItemGroupOrderedFactory.FILING_HISTORY_ID;
+import static uk.gov.companieshouse.itemhandler.kafka.ItemGroupOrderedFactory.FILING_HISTORY_TYPE;
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/itemhandler/kafka/OrderMessageDefaultConsumerIntegrationTest.java:67]: The statics imports should probably be before the non-static ones?

File: src/test/java/uk/gov/companieshouse/itemhandler/kafka/ItemGroupOrderedMessageProducerTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.ListenableFutureCallback;
+import uk.gov.companieshouse.itemgroupordered.ItemGroupOrdered;
+import uk.gov.companieshouse.itemhandler.itemsummary.ItemGroup;
+import uk.gov.companieshouse.itemhandler.model.OrderData;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/itemhandler/kafka/ItemGroupOrderedMessageProducerTest.java:21]: The static imports should be before non-static imports

File: src/test/java/uk/gov/companieshouse/itemhandler/kafka/ItemGroupOrderedFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.util.Collections;
+
+import static java.time.LocalDateTime.now;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/itemhandler/kafka/ItemGroupOrderedFactoryTest.java:23]: The static imports should be before non-static imports

File: src/test/java/uk/gov/companieshouse/itemhandler/kafka/ItemGroupOrderedAvroSerializerTest.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.itemgroupordered.ItemGroupOrdered;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/itemhandler/kafka/ItemGroupOrderedAvroSerializerTest.java:14]: The static imports should be before non-static imports

File: src/main/java/uk/gov/companieshouse/itemhandler/kafka/ItemGroupOrderedMessageProducer.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+import org.springframework.stereotype.Component;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.ListenableFutureCallback;
+import uk.gov.companieshouse.itemgroupordered.ItemGroupOrdered;
+import uk.gov.companieshouse.itemhandler.itemsummary.ItemGroup;
+import uk.gov.companieshouse.logging.Logger;
+
+import org.springframework.kafka.support.SendResult;
+
+import static uk.gov.companieshouse.itemhandler.logging.LoggingUtils.getLogMap;
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/itemhandler/kafka/ItemGroupOrderedMessageProducer.java:15]: The static imports should be before non-static imports

File: src/test/java/uk/gov/companieshouse/itemhandler/kafka/OrderRoutingIntegrationTest.java [hunk 6]
```diff
@@ -38,11 +38,11 @@
 import static java.util.concurrent.TimeUnit.SECONDS;
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.MatcherAssert.assertThat;
-
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.mockserver.model.HttpRequest.request;
 import static org.mockserver.model.HttpResponse.response;
+import static org.springframework.kafka.test.utils.KafkaTestUtils.getSingleRecord;
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/itemhandler/kafka/OrderRoutingIntegrationTest.java:45]: The static imports should be before non-static imports

**Review by hwilliams1 — CHANGES_REQUESTED** [review]
One formatting change in Javadoc comment, and some static imports that should be before the non-static imports but otherwise looks fine to me

**Comment by VaughanJackson123** [comment]
> One formatting change in Javadoc comment, and some static imports that should be before the non-static imports but otherwise looks fine to me

Thank you @hwilliams1 . Sorry, which Javadoc comment needs reformatting?

File: src/main/java/uk/gov/companieshouse/itemhandler/kafka/ItemGroupOrderedFactory.java [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+       }
+       return null;
+    }
+
+    /**
+     * Creates a suitable map of values representing copy filing history item options ready for use as part of a
+     * Kafka message.
+     *
+     * @param options {@link uk.gov.companieshouse.itemhandler.model.ItemOptions} the current copy item being processed
+     *                                                                           in the order
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/itemhandler/kafka/ItemGroupOrderedFactory.java:128]: This is the line in Javadoc which should be formatted

**Review by hwilliams1 — APPROVED** [review]

**Pull request merged**